src/eula.rs

```diff
@@ -1,116 +1,272 @@
-                trace!(
-                    "The '{}' path from the 'license-file' field in the package's \
-                     manifest (Cargo.toml) exists but it does not have a RTF file \
-                     extension.",
-                    license_file_path
-                );
-                Ok(Eula::Disabled)
-            }
-        } else if let Some(license_name) = package.license.as_ref() {
-            trace!("The 'license' field is specified in the package's manifest (Cargo.toml)");
-            debug!("license_name = {:?}", license_name);
-            if let Ok(template) = Template::from_str(license_name) {
-                trace!(
-                    "An embedded template for the '{}' license from the package's \
-                     manifest (Cargo.toml) exists.",
-                    license_name
-                );
-                Ok(Eula::Generate(template))
-            } else {
-                trace!(
-                    "The '{}' license from the package's manifest (Cargo.toml) is \
-                     unknown or an embedded template does not exist for it",
-                    license_name
-                );
-                Ok(Eula::Disabled)
-            }
-        } else {
-            trace!("The 'license' field is not specified in the package's manifest (Cargo.toml)");
-            Ok(Eula::Disabled)
-        }
-    }
-}
-
-impl fmt::Display for Eula {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Eula::CommandLine(ref path) => path.fmt(f),
-            Eula::Manifest(ref path) => path.fmt(f),
-            Eula::Generate(..) => write!(f, "{LICENSE_FILE_NAME}.{RTF_FILE_EXTENSION}"),
-            Eula::Disabled => write!(f, "Disabled"),
-        }
-    }
-}
+// Copyright (C) 2017 Christopher R. Field.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Error;
+use crate::Result;
+use crate::StoredPath;
+use crate::StoredPathBuf;
+use crate::Template;
+use crate::LICENSE_FILE_NAME;
+use crate::RTF_FILE_EXTENSION;
+
+use camino::Utf8Path;
+use camino::Utf8PathBuf;
+use log::trace;
+use log::warn;
+
+use std::str::FromStr;
+
+use cargo_metadata::Package;
+
+/// License info
+#[derive(Clone, Debug)]
+pub struct Licenses {
+    /// The license for the actual source code
+    ///
+    /// This likely will become/contain a Vec at some point,
+    /// since dual MIT/Apache wants to have two license files!
+    pub source_license: Option<License>,
```

Comment:
Can this just be called `source` and `end_user`? We already know these are licenses based on the type definition and being fields in the `Licenses` struct.